WooCommerce/Classes/ViewRelated/Orders/Order Creation/EditableOrderViewModel.swift

```diff
@@ -1255,6 +1266,42 @@ extension EditableOrderViewModel {
     func requestCameraAccess(onCompletion: @escaping ((Bool) -> Void)) {
         permissionChecker.requestAccess(for: .video, completionHandler: onCompletion)
     }
+
+    /// Attempts to add a Product to the current Order by SKU search
+    ///
+    func addScannedProductToOrder(barcode sku: String?, onCompletion: @escaping (Result<Void, Error>) -> Void) {
+        guard let sku = sku else {
+            return onCompletion(.failure(ScannerError.nilSKU))
+        }
+
+        mapFromSKUtoProductID(sku: sku) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case let .success(productID):
+                self.configureProductRowViewModels()
```

Comment:
What's this call for? Is it only required for tests? It seems odd to do it before adding the product, and I'd kind of expect it to be in the completion chain from setting the product on the order...

Reply:
> What's this call for? Is it only required for tests?  

TLDR;  As we're testing the contents of the ordersynchronizer now via the `currentOrderItems` computed var, I was able to remove this call on 350dc7e9a90d7c031c0e15e5f3da3e7583f06ef9 and still test the success case. This leaves us with a getter in production that is only used for testing, but IMHO is much better than having a random call in the scanner implementation, or to test productRows count.

Longer explanation: 

That's a good question, it took me quite a bit to understand what is happening, from what I understand:

Up until now we relied on counting the viewmodel's product rows when testing the EditableOrderViewModel. This made sense because the underlying logic for interacting with products relies on tapping them, which alters the state and triggers the different observers, updates state, etc, ... however by sending the products directly to the remote we seem to have some race condition in tests (perhaps also related the way we use MockStorage or we inject fake products in our test double) when we're expecting the updated values from the synchronizer's publishers and the contents of results controller: The notification seem to be sent before the variable itself is updated, as (the latest value in .sink is in willSet, not didSet), so we update the related methods by passing the out-of-date value.

This means we're late and the specific `ProductRowViewModel` fails to be created due to its coupling with fetching products from storage, as at the point we succeed to make our scanner call we notify the publisher but "we have not fetched yet" (we're accessing the old value, rather than the new), so the internal `allProducts` array is empty, which makes us return early from the different state that needs to be set for generating product rows. 

This call assured we're fetching from storage and the ProductRowViewmodels are set before we return from scanning. Since now we're not testing for product rows being rendered, but we test the contents of the sync order instead, we have removed this method call.

Reply:
Thanks for the explanation, and I agree that while the getter for tests only isn't brilliant, it's better than the alternative here!